### ISSUE TITLE
Need Code Request Generator

### ISSUE THREAD

**Issue by darkterminal** [usebruno/bruno#58]
Is Code Request Generator will available in the future?

**Comment by helloanoop** [comment]
Yes. This would be a great feature to support.
For the UX, We can start with adding another option in the dropdown called "Generate Code"
![image](https://user-images.githubusercontent.com/9350733/198997546-565e8fa5-9ff4-459d-87f5-00a0c5a129ea.png)

Its recommended to start supporting one language and framework first. Get the rough edges polished before supporting more languages. (Ex: Start with javascript and fetch)

**Comment by helloanoop** [comment]
For UX inspiration and pattern, we can use similar code that we have for `<EnvironmentSettings />` [here](https://github.com/usebruno/bruno/blob/main/packages/bruno-app/src/components/Environments/EnvironmentSettings/index.js)

![image](https://user-images.githubusercontent.com/9350733/198998111-4d6cda5a-0a91-42f1-9d6f-33c0d13737e1.png)

**Comment by fitiskin** [comment]
I'd like to contribute.

**Comment by helloanoop** [comment]
@fitiskin Sure, please checkout the contributing guide which will help you bring up bruno on your machine.
I am happy to help if you get blocked or answer any questions that you have.

**Comment by mark05e** [comment]
There are a few libraries that can help generate the code.

- https://github.com/Kong/httpsnippet
- https://github.com/postmanlabs/postman-code-generators
- https://github.com/curlconverter/curlconverter

**Comment by Gvinod1991** [comment]
I started working on this issue. I made some progress on UI part refer the screen shots below and will work on the code generation logic.

![bruno_generate_code_1](https://user-images.githubusercontent.com/13873668/200597951-c73d8c46-3794-43b3-b92b-9936334f442a.png)

![bruno_generate_code_2](https://user-images.githubusercontent.com/13873668/200597958-fd442592-6fcb-4916-b8f1-4e79397f1454.png)

![bruno_generate_code_3](https://user-images.githubusercontent.com/13873668/200597939-31f02e26-8900-4256-a34f-b04b6406f3d5.png)

**Comment by helloanoop** [comment]
This looks great @Gvinod1991 !

I would recommend to implement the `fetch` code generation first.
I would recommend to work on implementing a util function that takes the `item` object and converts it into a string like below which can be injected into  the code mirror editor'

I know this is an oversimplification :), and that a lot of nuance is there in getting the fetch codegen to work.

I wouldn't want to use postman/insomnia codegen lib (we are kind of competing against them 😉)
But would be willing to use any other generic codegen opensource lib if available

```javascript
// data to be sent to the POST request
let data = {
  title: "foo",
  body: "bar", 
  userId:1
}

let headers = [{
  "Content-type": "application/json,
  charset=UTF-8"
}]

fetch('https://jsonplaceholder.typicode.com/posts', {
  method: "POST",
  body: JSON.stringify(data),
  headers: headers
})
.then(response => response.json()) 
.then(json => console.log(json));
.catch(err => console.log(err));
```

**Comment by darkterminal** [comment]
I like first link as mention by @mark05e
> https://github.com/Kong/httpsnippet

**Comment by helloanoop** [comment]
Yup, agree with @darkterminal  (It does not make sense to reinvent the wheel, and write our own codegen)

I am ok on proceeding with https://github.com/Kong/httpsnippet
It's also shipped with MIT license

**Comment by helloanoop** [comment]
Code Generation is being implemented as a part of the Pro Edition of Bruno.
The Pro Edition code is not opensource.
Hence closing this issue.

**Comment by helloanoop** [comment]
@Beedhan Thanks for getting this done!

This is now available in `v0.17.0`
![image](https://github.com/usebruno/bruno/assets/9350733/25a2f848-2a2e-4024-9d20-00c175a62167)

**Comment by lizy-Leeeeee** [comment]
I think that's what most people want.

```
import { useQuery } from 'react-query'
import { genFetch } from 'bru-cli';
import searchReposBru from './SearchRepos.bru';

const searchRepos = genFetch(searchReposBru)
const { data } = useQuery('repos', searchRepos )
```
